soccer-records-model/src/main/java/cz/muni/fi/pa165/team/TeamMatchGoalRepositoryImpl.java

```diff
@@ -0,0 +1,77 @@
+package cz.muni.fi.pa165.team;
+
+import cz.muni.fi.pa165.team.exceptions.GoalNotFoundException;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.Assert;
+
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import java.util.Collection;
+import java.util.UUID;
+
+/**
+ * This class implements the TeamMatchGoalRepository interface.
+ *
+ * @author Tomas Smid <smid.thomas@gmail.com>
+ */
+@Repository
+public class TeamMatchGoalRepositoryImpl implements TeamMatchGoalRepository
+{
+
+    private EntityManager em;
+
+    public TeamMatchGoalRepositoryImpl(EntityManager em)
+    {
+        this.em = em;
+    }
+
+    @Override
+    public TeamMatchGoal getGoalById(final UUID goalId)
+    {
+        Assert.notNull(goalId, "Cannot search for a null goal id");
+
+        try {
+            return em.createQuery("SELECT g FROM TeamMatchGoal g WHERE g.id = :goalId", TeamMatchGoal.class)
+                .setParameter("goalId", goalId)
+                .getSingleResult();
+        } catch (NoResultException ex) {
+            throw new GoalNotFoundException(goalId, ex);
+        }
+    }
+
+    @Override
+    public Collection<TeamMatchGoal> findGoalByScorer(final UUID playerId)
```

Comment:
jo to jsem tam mel, pak jsem to zobecnoval, tak jsem si rikal, ze to bude lepsi, ale ok, predelam zpet